src/main.rs

```diff
@@ -464,6 +465,30 @@ fn search(
     Ok(Json(result))
 }
 
+#[get("/docs/<name>/<version>")]
+async fn get_package_docs(
+    db: &State<Database>,
+    name: String,
+    version: String,
+) -> Result<Redirect, Status> {
+    use forc_pub::file_uploader::pinata::ipfs_hash_to_docs_url;
+
+    let package_result =
+        db.transaction(|conn| conn.get_full_package_with_categories(name, version));
+
+    match package_result {
+        Ok(package_data) => {
+            if let Some(docs_hash) = package_data.package.docs_ipfs_hash {
+                let docs_url = ipfs_hash_to_docs_url(&docs_hash);
+                Ok(Redirect::to(docs_url))
+            } else {
+                Err(Status::NotFound)
+            }
+        }
+        Err(_) => Err(Status::NotFound),
```

Comment:
nit: Both database errors and missing packages return the same 404 status, which could make debugging harder.

```suggestion
            } else {
                // Package exists but has no documentation
                Err(Status::NotFound)
            }
        }
        Err(err) => {
            // Log the database error for debugging
            tracing::error!("Database error in get_package_docs: {:?}", err);
            Err(Status::NotFound)
        }
```